docs/getting-started/dockstore-tools.rst

```diff
@@ -1,28 +1,45 @@
 
 Register a Tool on Dockstore
 ============================
 
+.. contents::
+   :local:
+   :depth: 2
+
 Tutorial Goals
 --------------
 
+-  Learn about different ways to register a tool
 -  Register a tool on Dockstore
--  Learn about Quick registration vs Manual Registration
 -  Publish your tool
 
 Register Your Tool in Dockstore
 -------------------------------
+.. include:: /getting-started/github-apps/note--registration.rst
 
-Now that you have your ``Dockerfile``, ``Dockstore.cwl`` and/or
-``Dockstore.wdl`` in GitHub, have setup Quay.io to automatically build
+If you followed :doc:`Getting Started With CWL </getting-started/getting-started-with-cwl>`,
+you will now have your ``Dockerfile`` and ``Dockstore.cwl`` 
+in GitHub, have setup Quay.io to automatically build
 your Docker image, and have linked your accounts to Dockstore, it is
 time to register your tool. This tutorial will assume you are using CWL
-and/or WDL. Currently, Nextflow is not supported for tools on Dockstore.
+and/or WDL.
+
+.. note:: WDL tools are a legacy feature. Although you can still register WDL tools on Dockstore using the methods below, we are now encouraging people to :doc:`register WDLs as workflows </getting-started/dockstore-workflows>` instead. Tools are not supported at all for Galaxy or Nextflow; they can only be workflows.
+
+Option A: Registration with the Dockstore GitHub App (Recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+As of Release 1.12, we have a new method for registering tools that we
+recommend most people use, as it allows for automatic sychronization
+between Dockstore and GitHub. You can read more about it :doc:`in our docs about the Dockstore GitHub App </getting-started/github-apps/github-apps>`.
+
+.. I made the decision to not make this symmetric with the GitHub App-related section of dockstore-workflows.rst, because the differences between gh-tools and non-gh-tools are significant. Whenever possible, when people are learning about tools and the GH app, they should be taken to the chart breaking down the differences between gh-tools and non-gh-tools. I do not think embedding that chart here is a good idea as it will confuse users who are not used to the non-gh-tool standard. Just introduce one in this doc, and link out to the other standard at the top.
+
+.. With that being said, there is a potential to confuse users, as we do not have documentation that is labeled as being for registering a brand new tool onto Dockstore via the app, only converting old tools. I don't know what the best solution for this is. I think further improvements will be needed.
 
```

Comment:
It's grayed out on github, might be a code comment

Reply:
It's a comment, yes. RST does not quite support comments, but if you write
``.. foo``
Then it will basically act as a comment, provided it does not recognize "foo" as a keyword.